### ISSUE TITLE
nsq_to_file: multiple -lookupd-http-address

### ISSUE THREAD

**Issue by huahuahao2** [nsqio/nsq#731]
When passing multiple -lookupd-http-address to nsq_to_file, any of the lookupd became unreachable will causing nsq_to_file ERROR out.
`nsqd v0.3.7`

**Comment by jehiah** [comment]
@huahuahao2 can you be more specific about what you mean by `error out`? Does the process exit, or log and error and continue or ... ?

**Comment by huahuahao2** [comment]
From the log, here is what happened:
lookupd1 dies first, and nsq_to_file is still consuming data from the existing nsqd connects. But it cannot refresh topic list at that point, even the lookupd2 is working.
At some point today, nsq_to_file stopped for some reason. See the following log.
```
2016/04/01 14:40:02 INF    2 [topic1/nsq_to_file] stopping...
2016/04/01 14:40:02 INF    1 [topic2/nsq_to_file] stopping...
```
And when my nanny process tries to restart.
```
2016/04/01 14:40:50 ERROR: could not retrieve topic list: Get http://lookupd1:8198/topics: dial tcp 127.2.56.202:8198: i/o timeout
```
The process just exit.

**Comment by jehiah** [comment]
Thanks @huahuahao2 this does sound like a bug. nsq_to_file should not exit if one or more `nsqlookupd`'s are temporarily unavailable.

**Comment by mreiferson** [comment]
I'd like to see the rest of the logs if possible? This seems likely related to the "archive all topics" functionality, and wouldn't happen if you specified a topic.

**Comment by mreiferson** [comment]
I think this has since been fixed, probably in #936 or related changes
